src/java.base/share/classes/jdk/internal/jimage/ImageReader.java

```diff
@@ -92,146 +125,113 @@ private void requireOpen() {
-        requireOpen();
-        return reader.getHeader();
-    }
-
+    /**
+     * Releases a (possibly cached) {@link ByteBuffer} obtained via
+     * {@link #getResourceBuffer(Node)}.
+     *
+     * <p>Note that no testing is performed to check whether the buffer about
+     * to be released actually came from a call to {@code getResourceBuffer()}.
+     */
     public static void releaseByteBuffer(ByteBuffer buffer) {
         BasicImageReader.releaseByteBuffer(buffer);
     }
 
-    public String getName() {
-        requireOpen();
-        return reader.getName();
-    }
-
-    public ByteOrder getByteOrder() {
+    /**
+     * Returns the content of a resource node in a possibly cached byte buffer.
+     * Callers of this method must call {@link #releaseByteBuffer(ByteBuffer)}
+     * when they are finished with it.
+     */
+    public ByteBuffer getResourceBuffer(Node node) {
         requireOpen();
-        return reader.getByteOrder();
-    }
-
-    public Path getImagePath() {
-        requireOpen();
-        return reader.getImagePath();
-    }
-
-    public ImageStringsReader getStrings() {
-        requireOpen();
-        return reader.getStrings();
-    }
-
-    public ImageLocation findLocation(String mn, String rn) {
-        requireOpen();
-        return reader.findLocation(mn, rn);
-    }
-
-    public boolean verifyLocation(String mn, String rn) {
-        requireOpen();
-        return reader.verifyLocation(mn, rn);
-    }
-
-    public ImageLocation findLocation(String name) {
-        requireOpen();
-        return reader.findLocation(name);
-    }
-
-    public String[] getEntryNames() {
-        requireOpen();
-        return reader.getEntryNames();
-    }
-
-    public String[] getModuleNames() {
-        requireOpen();
-        int off = "/modules/".length();
-        return reader.findNode("/modules")
-                     .getChildren()
-                     .stream()
-                     .map(Node::getNameString)
-                     .map(s -> s.substring(off, s.length()))
-                     .toArray(String[]::new);
-    }
-
-    public long[] getAttributes(int offset) {
-        requireOpen();
-        return reader.getAttributes(offset);
-    }
-
-    public String getString(int offset) {
-        requireOpen();
-        return reader.getString(offset);
-    }
-
-    public byte[] getResource(String name) {
-        requireOpen();
-        return reader.getResource(name);
-    }
-
-    public byte[] getResource(ImageLocation loc) {
-        requireOpen();
-        return reader.getResource(loc);
-    }
-
-    public ByteBuffer getResourceBuffer(ImageLocation loc) {
-        requireOpen();
-        return reader.getResourceBuffer(loc);
-    }
-
-    public InputStream getResourceStream(ImageLocation loc) {
-        requireOpen();
-        return reader.getResourceStream(loc);
+        if (!node.isResource()) {
+            throw new IllegalStateException("Not a resource node: " + node);
```

Comment:
Should this throw `IllegalArgumentException` instead of `IllegalStateException`?